numpy/_core/src/multiarray/calculation.c

```diff
@@ -700,7 +700,7 @@ PyArray_Round(PyArrayObject *a, int decimals, PyArrayObject *out)
  finish:
     Py_DECREF(f);
     Py_DECREF(out);
-    if (ret_int) {
+    if (ret_int && ret != NULL) {
```

Comment:
Sorry, unrelated and a bit annoying to trigger.  But the deprecations triggered the bug...